ugs-platform/ugs-platform-ugscore/src/main/java/com/willwinder/ugs/nbp/core/toolbars/Bundle.properties

```diff
@@ -0,0 +1,4 @@
+mainWindow.swing.firmware.toolbarTitle = Firmware selection toolbar
```

Comment:
@joacimb I also tried to set everything in the constructor and started several discussions on the NetBeans mailing list about different localization subjects like this. I think the annotations are package scanned by a separate service to register everything, so it really doesn't work this way. So everything needs to be initialized in a static final variable so that it can be annotated outside the class, which is what I use `LocalizingService.java`. It declares a bunch of static final strings.  In `JogControlTopComponent` for instance is this:
```
@ActionID(category = LocalizingService.JogControlCategory, id = LocalizingService.JogControlActionId)
@ActionReference(path = LocalizingService.JogControlWindowPath)
```

Which properly registers the different modules into the menu. It would be nice if annotations could be set dynamically, but I wasn't able to find a way to do it.

Looking at it a little more, the toolbar instances might need to be updated in the `LocalizingService` constructor the same way menu items are updated.